substrate/client/network/transactions/src/lib.rs

```diff
@@ -345,6 +357,9 @@ where
 			},
 			NotificationEvent::NotificationStreamClosed { peer } => {
 				let _peer = self.peers.remove(&peer);
+				if _peer.is_none() {
+					panic!("peer doesn't exist {peer:?}");
```

Comment:
```suggestion
					log::error!(target: "transactions", "Cannot remove unknown peer {peer:?}");
```
May be it's enough to have `debug_assert!`?

Reply:
This is actually not needed at all, there was a bug in `Peerstore` which caused `TransactionsHandler` incorrectly reject a notification stream. I'll restore the original code.